### PR TITLE
rubocop issues - Cleaning up

### PR DESCRIPTION
**Pull request by shleeable**
Starting to get more into the code and make actual changes... all of these make sense to me - but as with my other PRs, you might have personal choice to keep using different code styles.

### CODE REVIEW

File: config/initializers/trusted_proxies.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ def trusted_proxy?(ip)
       if Rails.application.config.action_dispatch.trusted_proxies.nil?
         super
       else
-        Rails.application.config.action_dispatch.trusted_proxies.any? { |proxy| proxy === ip }
+        Rails.application.config.action_dispatch.trusted_proxies.any? { |proxy| proxy.include?(ip) }
```
Comment (Gargron) [inline config/initializers/trusted_proxies.rb]: Why? This is not right. `proxy` is IPAddr, not an array.

File: config/initializers/open_uri_redirection.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 module OpenURI
   def self.redirectable?(uri1, uri2) # :nodoc:
-    uri1.scheme.downcase == uri2.scheme.downcase ||
+    uri1.scheme.casecmp(uri2.scheme) ||
```
Comment (Gargron) [inline config/initializers/open_uri_redirection.rb]: This is wrong. Output of casecmp is -1, 0, 1, not boolean. Need to `casecmp(...).zero?` for correct behaviour

File: app/models/concerns/account_interactions.rb [hunk 3]
```diff
@@ -209,8 +209,6 @@ def lists_for_local_distribution
          .where('users.current_sign_in_at > ?', User::ACTIVE_DURATION.ago)
   end
 
-  private
```
Comment (Gargron) [inline app/models/concerns/account_interactions.rb]: Why?

File: app/lib/settings/scoped_settings.rb [hunk 4]
```diff
@@ -47,7 +47,6 @@ def []=(key, value)
       record.update!(value: value)
 
       Rails.cache.write(Setting.cache_key(key, @object), value)
-      value
```
Comment (Gargron) [inline app/lib/settings/scoped_settings.rb]: `Rails.cache.write` does not have a documented return value. I would need some reasoning for leaving  this function without a return value.

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/lib/settings/scoped_settings.rb (hunk 4 above)
Comment (ashleyhull-versent) [reply app/lib/settings/scoped_settings.rb]: ok, the fact this value is written and used as the return was lost on me.. It seems to alert in rubocop as "Variable `value` used in void context".

https://rubocop.readthedocs.io/en/latest/cops_lint/#lintvoid

File: app/models/concerns/account_interactions.rb (hunk 3 above)
Comment (ashleyhull-versent) [reply app/models/concerns/account_interactions.rb]: ok. I was certain this is just a typo... why is the private just sitting there between these two defs?

I'll revert this change.

File: config/initializers/open_uri_redirection.rb (hunk 2 above)
Comment (ashleyhull-versent) [reply config/initializers/open_uri_redirection.rb]: thanks for the catch... I actually meant to .zero it

File: config/initializers/trusted_proxies.rb (hunk 1 above)
Comment (ashleyhull-versent) [reply config/initializers/trusted_proxies.rb]: sorry. I miss-read this.

**Comment by ashleyhull-versent** [comment]
@Gargron uncommited those mistakes.

File: config/initializers/open_uri_redirection.rb [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 module OpenURI
   def self.redirectable?(uri1, uri2) # :nodoc:
-    uri1.scheme.downcase == uri2.scheme.downcase ||
+    uri1.scheme.casecmp(uri2.scheme).zero? ||
```
Comment (ashleyhull-versent) [inline config/initializers/open_uri_redirection.rb]: This looks correct to me - but please confirm.

**Review by Gargron — APPROVED** [review]

**Pull request merged**